lib/common.bash

```diff
@@ -22,65 +22,116 @@ info() {
-	PROXY_VERSION=$(awk '/^\[Proxy\]$/ {foundit=1} /^  Version =/ { if (foundit==1) {print $5; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+			SHIM_PATH=$(awk '/^\[Shim\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+			SHIM_VERSION=$(awk '/^\[Shim\]$/ {foundit=1} /^  Version =/ { if (foundit==1) {$1=$2=""; print $0; foundit=0} } ' <<< "$toml" | sed 's/"//g')
 
-	HYPERVISOR_PATH=$(awk '/^\[Hypervisor\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
-	HYPERVISOR_VERSION=$(awk '/^\[Hypervisor\]$/ {foundit=1} /^  Version =/ { if (foundit==1) {$1=$2=""; print $0; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+			PROXY_PATH=$(awk '/^\[Proxy\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+			PROXY_VERSION=$(awk '/^\[Proxy\]$/ {foundit=1} /^  Version =/ { if (foundit==1) {print $5; foundit=0} } ' <<< "$toml" | sed 's/"//g')
 
-	INITRD_PATH=$(awk '/^\[Initrd\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+			HYPERVISOR_PATH=$(awk '/^\[Hypervisor\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+			HYPERVISOR_VERSION=$(awk '/^\[Hypervisor\]$/ {foundit=1} /^  Version =/ { if (foundit==1) {$1=$2=""; print $0; foundit=0} } ' <<< "$toml" | sed 's/"//g')
 
-	NETMON_PATH=$(awk '/^\[Netmon\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+			INITRD_PATH=$(awk '/^\[Initrd\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+
+			NETMON_PATH=$(awk '/^\[Netmon\]$/ {foundit=1} /^  Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
+			return 0
+		fi
+	fi
+
+	# We have not found a command with a 'kata-env' option we can run. Set up some
+	# default values.
+	# We could be more diligent here and search for each individual component,
+	# but if the runtime cannot tell us the exact details it is configured for then
+	# we would be guessing anyway - so, set some defaults that may be true and give
+	# strong hints that we 'made them up'.
+	info "Runtime environment not found - setting defaults"
+	RUNTIME_CONFIG_PATH="/usr/share/defaults/kata-containers/configuration.toml"
+	RUNTIME_VERSION="0.0.0"
+	RUNTIME_COMMIT="unknown"
+	# If docker is broken, disabled or not installed then we may not get a runtime
+	# path from it...
+	if [ -z "$RUNTIME_PATH" ]; then
+		RUNTIME_PATH="/usr/bin/kata-runtime"
+	else
+		RUNTIME_PATH="$rpath"
+	fi
+	SHIM_PATH="/usr/libexec/kata-containers/kata-shim"
+	SHIM_VERSION="0.0.0"
+	PROXY_PATH="/usr/libexec/kata-containers/kata-proxy"
+	PROXY_VERSION="0.0.0"
+	if [ "$KATA_HYPERVISOR" == firecracker ]; then
+		HYPERVISOR_PATH="/usr/bin/firecracker"
+	 else
+		HYPERVISOR_PATH="/usr/bin/qemu-system-x86_64"
```

Comment:
what about not hardcoding the `x86_64` arch and use the `kata-arch.sh` here, something like:
```
HYPERVISOR_ARCH=$(${cidir}/kata-arch.sh -d)
HYPERVISOR_PATH="/usr/bin/qemu-system-${HYPERVISOR_ARCH}"
```

Reply:
good idea, let me update.